lib/fetch/index.js

```diff
@@ -1759,7 +1759,12 @@ async function httpNetworkFetch (
       // 1. If connection uses HTTP/2, then transmit an RST_STREAM frame.
       fetchParams.controller.connection.destroy()
 
-      // 2. Return the appropriate network error for fetchParams.
+      // 2. If error is ProxyAgent related, then pass it.
+      if (err.message === 'Proxy response !== 200 when HTTP Tunneling') {
```

Comment:
```suggestion
      if (err.code === 'UND_ERR_ABORTED') {
```

Reply:
In my code, I check whether the request was aborted exactly because of a proxy error. And there is no need for a second check that the request has been aborted, there is already one.
https://github.com/nodejs/undici/blob/00f40cdc8b102053f2c6f8fac9b5bf09df4c4fe3/lib/fetch/index.js#L1758

Reply:
checking error messages is bound to cause issues

Reply:
Then I suggest adding a new UndiciError subclass - ProxyError. There is no other way to separate the errors related to the ProxyAgent.